### DISCUSSION TITLE
Best way to logout when using Oauth2 and cookies

### DISCUSSION THREAD

**Discussion by bert2002** [fastapi-users/fastapi-users#1234] [Q&A]
Hi,
what is the best way to logout a user from its cookie session? I am using Oauth2 with cookie transport and redis strategy. Once the lifetime expired the session is invalid and the user gets logged out automatically, but is there an integrated way to logout a user?

When I check the documentation I see that the [Cookie](https://fastapi-users.github.io/fastapi-users/11.0/configuration/authentication/transports/cookie/) should have a logout method, but its not created automatically.

When I enable `get_auth_router`:
```
app.include_router(
  fastapi_users.get_auth_router(auth_backend), prefix="/auth/jwt", tags=["auth"]
)
```
I get a `logout` method that I can use to logout (and it works), but it also creates a `login` endpoint for username/password which I dont want. 

Is it possible to get only the `logout` route as GET request? 

Thanks,
bert

**Comment by frankie567** [comment]
Yes, the auth router directly comes with both `login` and `logout` routes.

You can either copy/paste the `logout` route in your app (it's very short), or you can go through the defined routes in the router and filter out login. A bit hacky but it works:

```py
auth_router = fastapi_users.get_auth_router(auth_backend)
auth_router.routes = [route for route in auth_router.routes if route.name != f"auth:{auth_backend.name}.login"]
app.include_router(auth_router, prefix="/auth/jwt", tags=["auth"])
```

**Reply by bert2002** [reply]
I gave it a try to copy/paste the `logout` route, but it fails when I want to import `BaseAuthentication`:

```
from fastapi_users.authentication import Authenticator, BaseAuthentication
```
and I am getting:
```
from fastapi_users.authentication import Authenticator, BaseAuthentication
   ImportError: cannot import name 'BaseAuthentication' from 'fastapi_users.authentication' (/usr/local/lib/python3.11/site-packages/fastapi_users/authentication/__init__.py)
```
Thats happening in my `main.py`.

Your hacky workaround does work though :)
Btw. Is there a reason why its a `POST` request?
